src/screens/explore/CardById/CardById.tsx

```diff
@@ -0,0 +1,217 @@
+import { NavigationProp, RouteProp, useNavigation } from '@react-navigation/native';
+import { StackNavigationProp } from '@react-navigation/stack';
+import { StatusBar } from 'expo-status-bar';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import {
+  RefreshControl,
+  SafeAreaView,
+  ScrollView,
+  TouchableOpacity,
+  View,
+  Image,
+} from 'react-native';
+import { useSelector } from 'react-redux';
+import { Button, ButtonType, Icon, ImageCarousel, Text, ThemedView } from 'src/components';
+import { ScreenTemplate } from 'src/components/templates';
+import { RootStackParamList } from 'src/navigation';
+import { useAppDispatch } from 'src/store';
+import { getIsDarkMode } from 'src/store/selectors';
+import { AsyncThunks } from 'src/store/thunks';
+import { BLACK, BUTTON_SIZES, LEVEL_1, TOMATO, WHITE } from 'src/styles';
+import { AccommodationFullView, IconName } from 'src/types';
+
+import { styles } from './CardById.styles';
+import { AMENITIES_CHIP_DATA, DEFAULT_ACCOMMODATION_VIEW, formatDate } from './CardById.utils';
+
+type CardByIdNavigationProp = StackNavigationProp<RootStackParamList, 'CardById'>;
+type CardByIdRouteProp = RouteProp<RootStackParamList, 'CardById'>;
+
+type CardByIdProps = {
+  navigation: CardByIdNavigationProp;
+  route: CardByIdRouteProp;
+};
+
+const CardById = ({ route }: CardByIdProps) => {
+  const acccomodationId = route.params?.accomodationId;
+  const dispatch = useAppDispatch();
+  const navigation = useNavigation<NavigationProp<RootStackParamList>>();
+  const theme = useSelector(getIsDarkMode);
+  const [cardData, setCardData] = useState<AccommodationFullView>(DEFAULT_ACCOMMODATION_VIEW);
+  const [heartPressed, setHeartPressed] = useState<boolean>(false);
+  const [refreshing, setRefreshing] = useState<boolean>(false);
+
+  const mappedMedia = cardData.media.map((media) => media.imageUrl);
+
+  const handleHeartPress = () => {
+    setHeartPressed(!heartPressed);
+  };
+
+  const getCardData = useCallback(async () => {
+    const response = await dispatch(AsyncThunks.getAccommodation(acccomodationId));
+    if (response.payload?.success) {
```

Comment:
when you use `redux-toolkit`, you don't need to use `useState` hook as all global state managers let you do the same, + you already have `accomodationSlice`. Just import a dedicated selector to get the details of the accommodation from `accomodationSlice`

Reply:
fixed